diff --git a/Makefile.am b/Makefile.am
index e060b68..2e041af 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -57,12 +57,16 @@ libminidjvu_mod_settings_la_SOURCES = \
  tools/settings-reader/AppOptions.cpp tools/settings-reader/AppOptions.h		\
  tools/settings-reader/SettingsReaderAdapter.cpp
 
-bin_PROGRAMS = minidjvu-mod
+bin_PROGRAMS = minidjvu-mod djvudict
 
 minidjvu_mod_SOURCES = tools/minidjvu-mod.c
 
 minidjvu_mod_LDADD = libminidjvu-mod.la libminidjvu-mod-settings.la
 
+djvudict_SOURCES = tools/djvudict.cpp tools/bsdecoder.cpp tools/djvudirreader.cpp tools/jb2dumper.cpp tools/sqlstorage.cpp
+
+djvudict_LDADD = libminidjvu-mod.la
+
 minidjvu-mod.pc:
 	echo 'prefix=$(prefix)'			>  $@
 	echo 'exec_prefix=$(exec_prefix)'	>> $@
diff --git a/configure.ac b/configure.ac
index 878da64..3a99f07 100644
--- a/configure.ac
+++ b/configure.ac
@@ -34,6 +34,7 @@ AC_CHECK_LIB(z, inflate)
 AC_CHECK_LIB(jpeg, jpeg_destroy_decompress)
 AC_CHECK_LIB(tiff, TIFFOpen)
 AC_CHECK_LIB(jemalloc,malloc)
+AC_CHECK_LIB(sqlite3, sqlite3_open)
 # Check for OpenMP
 AC_OPENMP
 
diff --git a/src/jb2/jb2coder.h b/src/jb2/jb2coder.h
index 89f61fb..5aed23d 100644
--- a/src/jb2/jb2coder.h
+++ b/src/jb2/jb2coder.h
@@ -81,7 +81,7 @@ struct JB2Decoder : JB2Coder, JB2BitmapDecoder
 
     void reset(); // resets numcontexts as required by "reset" record
 
-    private:
+    //private:
         void decode_character_position(int32 &x, int32 &y, int32 w, int32 h);
 };
 
diff --git a/src/jb2/zp.h b/src/jb2/zp.h
index 82d1352..49a6a55 100644
--- a/src/jb2/zp.h
+++ b/src/jb2/zp.h
@@ -119,8 +119,9 @@ class ZPDecoder
         Bit decode_without_context();
         Bit decode(ZPBitContext &);
         int32 decode(ZPNumContext &);
-    private:
+
         FILE *file;
+    private:
         uint32 a, code, fence, buffer;
         int32 bytes_left;
         unsigned char byte, scount, delay;
