diff --git a/src/jb2/jb2coder.h b/src/jb2/jb2coder.h
index 89f61fb..af8d63d 100644
--- a/src/jb2/jb2coder.h
+++ b/src/jb2/jb2coder.h
@@ -9,6 +9,7 @@
 
 // struct JB2Rect - a simple rectangle class {{{
 
+
 struct JB2Rect
 {
     int left, top, width, height;
@@ -81,7 +82,7 @@ struct JB2Decoder : JB2Coder, JB2BitmapDecoder
 
     void reset(); // resets numcontexts as required by "reset" record
 
-    private:
+//    private:
         void decode_character_position(int32 &x, int32 &y, int32 w, int32 h);
 };
 
